share/spice/airlines/airlines.js

```diff
@@ -49,16 +49,64 @@
 
     env.ddg_spice_airlines = function(api_result) {
 
-        // store the user's airline and city search information
-        var queriedAirlines =  null,
-            sourceCity = null,
-            destinationCity = null;
-
-        // Check if we have anything returned.
+        // Check if FlightStats returned anything
         if (!api_result || !api_result.flightStatuses || !api_result.appendix || !api_result.appendix.airlines) {
             return;
         }
 
+        // extract parameters from the Flightstats request
+        var script = $('[src*="/js/spice/airlines/"]')[0];
+        var source = $(script).attr("src");
+        var callParameters = decodeURIComponent(source).split('/');
+        
+        // determine the original date that we made an upstream request
+        var originalRequestDayOfMonth = callParameters[8];
+        var originalRequestHour = callParameters[9];
+
+        // for every API request, we make an additional request to 
+        // grab results for late-night arrivals from the night before and 
+        // red eyes for the next day (relative to UTC)
+        //
+        // more broadly, this allows us to grab a larger window of flights
+        var apiResponseDate = new Date(Date.UTC(api_result.request.date.year,
+                                       api_result.request.date.month-1,
+                                       api_result.request.date.day,
+                                       originalRequestHour, 0, 0));
+
+        // if this response is for the original same-day request...
+        if (originalRequestDayOfMonth == apiResponseDate.getUTCDate()) {
+
+            var apiRequestDate;
+
+            //  and the current hour is before 3AM, get the previous day's flights as well
+            //  (so we can show arrivals for late-night flights)
+            if (originalRequestHour < 3) {
```

Comment:
We may want to hold on to a list of possible fallbacks in case the 2nd API call has nothing useful.

For example "flight EgyptAir 995" we currently show a result on the live site for this but not with these changes because it landed over 3 hours ago. However there's nothing else coming so we should probably still show this.